### PR TITLE
Nifti cleanup

### PR DESCRIPTION
**Pull request by jdtournier**
Follows on from discussion on #1318

### CODE REVIEW

File: core/header.cpp [hunk 1]
```diff
@@ -465,6 +465,12 @@ namespace MR
       WARN ("transform matrix contains invalid entries - resetting to sane defaults");
       transform() = Transform::get_default (*this);
     }
+    else {
```
Comment (jdtournier) [inline core/header.cpp]: This bit is not strictly related to this, so might be best not to include... We'd had discussions in the past as to how best to handle non-orthogonal transforms - I guess this might be one way...

**Comment by jdtournier** [comment]
OK, this has thrown up another issue: exceptions thrown by threads in ThreadedLoops were not caught - although the error was displayed. This means that just about all of the tests passed when they shouldn't have, apart from a single lone failure. It's only because of that failure I went to have a look at the log and realised there should have been more... 

I've fixed that, and I'll go and figure out why the others are failing. So far it looks like mismatched transform matrices...

**Comment by jdtournier** [comment]
OK, hopefully that'll fix it...

**Comment by jdtournier** [comment]
@Lestropie, can you give this the thumbs-up when you can? This needs merging ASAP if it's to make it onto the laptops. It's good to go as far as I can tell...

**Comment by Lestropie** [comment]
`Thread::run()` is also used in `dirflip`, `dirgen`, `dirsplit` and `tckglobal`, so a similar effect could occur there (though `dirgen` is the only one of those with a test).

File: core/header.cpp (hunk 1 above)
Comment (Lestropie) [reply core/header.cpp]: I seem to recall Dave having commented proudly at some point about being able to apply affine transformations directly to the image header and being able to operate on those images / even open them in `mrview` without necessitating image regridding. So I don't think this is something that should be "slipped in" without more careful consideration.

**Comment by jdtournier** [comment]
> Thread::run() is also used in dirflip, dirgen, dirsplit and tckglobal, so a similar effect could occur there (though dirgen is the only one of those with a test).

Good catch. Let's make those changes on `dev` for now, they're not so critical. 

About the rigid-body transform:

> I seem to recall Dave having commented proudly at some point about being able to apply affine transformations directly to the image header and being able to operate on those images / even open them in mrview without necessitating image regridding. So I don't think this is something that should be "slipped in" without more careful consideration. 

Agree with removing the rigid body test from this PR, it does need further discussion. But in general, I'm not sure I agree that we should allow non-rigid transforms... I'll open another issue for that discussion.
